diesel/src/pg/types/geometric.rs

```diff
@@ -0,0 +1,62 @@
+//! Support for Geometric types under PostgreSQL.
+
+use byteorder::{NetworkEndian, ReadBytesExt, WriteBytesExt};
+use std::error::Error;
+use std::io::prelude::*;
+
+use deserialize::{self, FromSql};
+use expression::AsExpression;
+use expression::bound::Bound;
+use pg::Pg;
+use serialize::{self, Output, ToSql, IsNull};
+use sql_types::{Point};
+
+/// Point is represented in Postgres as a tuple of 64 bit floating point values (x, y).  This
+/// struct is a dumb wrapper type, meant only to indicate the tuple's meaning.
+#[derive(Debug, Clone, PartialEq, Copy, FromSqlRow)]
+pub struct PgPoint(pub f64, pub f64);
+
+impl<'a> AsExpression<Point> for &'a PgPoint {
+    type Expression = Bound<Point, &'a PgPoint>;
+
+    fn as_expression(self) -> Self::Expression {
+        Bound::new(self)
+    }
+}
+
+impl FromSql<Point, Pg> for PgPoint {
+    fn from_sql(bytes: Option<&[u8]>) -> deserialize::Result<Self> {
+        let mut bytes = not_none!(bytes);
```

Comment:
Do I need a debug_assert! counting the number of bytes I expect (16bytes for point)? Some types have it, some don't.

Reply:
No, we include them on integers because it's common to accidentally end up with the wrong type. That's less of a concer here.